src/pcp.rs

```diff
@@ -513,6 +518,13 @@ where
             let proof_data = &pf.data[proof_len..proof_len + next_len];
             proof_len += next_len;
 
+            // Make sure the query randomness isn't a root of unity. Evaluating the gadget
+            // polynomial at any of these points would be a privacy violation, since these points
+            // were used by the prover to construct the wire polynomials.
+            if query_rand[idx].pow(F::Integer::try_from(m).unwrap()) == F::one() {
+                return Err(PcpError::QueryRandInvalid);
+            }
```

Comment:
nit: we could move this check up a few lines so it comes right after `m` is initially computed.

Reply:
Done.